core/blockchain.go

```diff
@@ -1022,11 +1022,6 @@ func (bc *BlockChain) SnapSyncCommitHead(hash common.Hash) error {
 	return nil
 }
 
-// StateAtWithSharedPool returns a new mutable state based on a particular point in time with sharedStorage
-func (bc *BlockChain) StateAtWithSharedPool(root common.Hash) (*state.StateDB, error) {
-	return state.NewWithSharedPool(root, bc.stateCache, bc.snaps)
```

Comment:
Does the `NewWithSharedPool` used in the `InsertChain` function need to be removed?

https://github.com/bnb-chain/bsc/blob/f28b98a9947b52816714f90326b2545b08606b1a/core/blockchain.go#L2013

Reply:
> Does the `NewWithSharedPool` used in the `InsertChain` function need to be removed?
> 
> https://github.com/bnb-chain/bsc/blob/f28b98a9947b52816714f90326b2545b08606b1a/core/blockchain.go#L2013

I am afraid not, `InsertChain` needs make sure backward compatibility, like the `hertzFix` case, so we can not remove the sharedPool logic from `InsertChain`